packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2179,7 +2177,6 @@ class StatefulElement extends ComponentElement {
         'that all the resources used by the widget are fully released.'
       );
     });
-    assert(!dirty); // See BuildableElement.unmount for why this is important.
```

Comment:
There isn't a BuildableElement.unmount() method (anymore) so I'm not sure why this was important. 

On the face of it, it makes sense that we wouldn't care if an element was dirty when we're about to drop it.